File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        int ans = 0, curr = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(') {
                curr++;
                ans = Math.max(ans, curr);
            }
            if (s.charAt(i) == ')') {
                curr--;
            }
        }
        return ans;
    }
}

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.py
class Solution(object):
    def maxDepth(self, s):
        """
        :type s: str
        :rtype: int
        """
        ans, curr = 0, 0
        for c in s:
            if c == '(':
                curr += 1
                ans = max(ans, curr)
            if c == ')':
                curr -= 1
        return ans